codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/generator/awssdk/v2/ToNativeAwsV2.java

```diff
@@ -608,19 +626,37 @@ protected MethodSpec errorOpaqueWithText() {
         VAR_INPUT,
         Dafny.datatypeDeconstructor("obj")
       )
-      // If obj is ANY Exception
+      // If obj is A RuntimeException
       .nextControlFlow(
         "else if ($L.$L instanceof $T)",
         VAR_INPUT,
         Dafny.datatypeDeconstructor("obj"),
-        Exception.class
+        RuntimeException.class
       )
       .addStatement(
         "return ($T) $L.$L",
         RuntimeException.class,
         VAR_INPUT,
         Dafny.datatypeDeconstructor("obj")
       )
+      // If obj is A Throwable
+      .nextControlFlow(
+        "else if ($L.$L instanceof $T)",
+        VAR_INPUT,
+        Dafny.datatypeDeconstructor("obj"),
+        Throwable.class
+      )
+      .addStatement(
+        """
+          return new RuntimeException(
+            String.format(
+              "Unknown error thrown while calling AWS. %s",
```

Comment:
Look at this: https://github.com/smithy-lang/smithy-dafny/blob/e1afcf0d6b126334488b2c8d24f5b40374c4e716/codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/generator/awssdk/v2/ToNativeAwsV2.java#L594

You can use that to tell the customer what service we are talking about,
since this might get wrapped up in other things.